components/time-picker/time-picker.component.ts

```diff
@@ -240,8 +240,11 @@ export class NzTimePickerComponent implements ControlValueAccessor, OnInit, Afte
     }
   }
 
-  onFocus(value: boolean): void {
-    this.focused = value;
+  onFocusChange(event: FocusEvent): void {
+    this.focused = event.type === 'focus';
+    if (!this.focused) {
+      this.close();
+    }
```

Comment:
```suggestion
    if (!value) {
      this.checkAndClose();
    }
```
The `checkAndClose` should contains two situations:
- If time is valid should change value and close, ref `onKeyupEnter()`;
- If time is invalid should reset to last value and close, ref `onKeyupEsc()`;

and these should be added to the tests.

Reply:
Thanks for the feedback.

This validation is actually already done in the onInputChange() method so I think I can get away with just adding setCurrentValueAndClose() that is already being used in the onClickOutside().

There is only one thing I'm not sure about. When you enter an invalid date or nothing, the value will change to the current dateTime. Is this intended behaviour? Because this means that you can't empty the field without clicking the x inside the component.